### ISSUE TITLE
Border StrokeShape should be Geometry, not a Shape

### ISSUE THREAD

**Issue by vanka78bg** [dotnet/maui#3558]
### Description

This is an API design oversight, I believe. The `Border` component has properties, such as `Stroke`, `StrokeThickness`, `Fill` etc. The border shape is specified via the `StrokeShape` property of type `IShape`. The implementing `Shape` class also has properties, such as `Stroke`, `StrokeThickness`, `Fill` etc. This is really confusing, because it is not known which set of properties is respected.

In my humble opinion, the correct type of the stroke shape should be `Geometry`, similar to what we use for the `Clip` property. In fact, it would be nice to clip the inner contents of the border to that `Geometry`. For consistency, the property should be renamed to `StrokeGeometry` as well.

In addition to this, I believe it is not very convenient to require specifying the border shape always, to see any border on the screen. If the border shape is not specified, it should default to a rectangular border, as this is the most common use case.

### (Public) API Changes

1. Remove or make obsolete the `StrokeShape` property of the `Border` class.
2. Add a new property `StrokeGeometry` of type `Geometry` to the `Border` class.

### Usage Scenarios

```csharp
var border = new Border
{
    Stroke = new SolidColorBrush(Colors.Red),
    StrokeGeometry = new RectangleGeometry(),
    StrokeThickness = 5
};
```

```xaml
<Border Stroke="Red" StrokeThickness=5>
    <Border.StrokeGeometry>
        <RectangleGeometry />
    </Border.StrokeGeometry>
</Border>
```

### Backward Compatibility

If the original `StrokeShape` property is made obsolete, then backward compatibility can be maintained. I vote for introducing a breaking change by removing the `StrokeShape` property and replacing it with the new `StrokeGeometry` property. Having a fresh new framework with some legacy APIs from day one is not good in my books.

### Difficulty

Medium

**Comment by vanka78bg** [comment]
After investigating the matter some more, I have discovered that both `Shape` and `Geometry` classes implement the `IShape` interface. Setting the `StrokeShape` property to an instance derived from `Shape` renders the border correctly, while setting it to an instance derived from `Geometry` does not render any border at all. Now I do not know whether this is a bug, or just the API is letting me to assign values of unsupported types. Anyway, I am more convinced this API needs some serious rethinking.

I would suggest having two interfaces: `IGeometry` and `IShape`, as these interfaces are intended to be consumed by the `Handler`s. The `IShape` interface can inherit form `IGeometry`. Now it would be possible to expose all these `Stroke`, `StrokeThickness`, `Fill` etc. properties from the `IShape` interface. This way the `Handler`s would not have to cast `IShape` to `Shape` every time to get access to them.

**Comment by vanka78bg** [comment]
Another thing to note: currently it is not possible to set a rectangular shape in XAML. For instance, the following XAML code snippet would not work:

```xaml
<Border>
    <Border.StrokeShape>
        <Rectangle />
    </Border.StrokeShape>
</ Border>
```

There is a naming conflict between `Microsoft.Maui.Graphics.Rectangle` and `Microsoft.Maui.Controls.Shapes.Rectangle`. Apparently, the XAML parser cannot determine which type to use, so it tries to set the wrong one. This is rather inconvenient, because `Rectangle` is the most common border type. Changing the property type to `Geometry` would allows us to specify it like this:

```xaml
<Border>
    <Border.StrokeShape>
        <RectangleGeometry />
    </Border.StrokeShape>
</ Border>
```

**Comment by shimingsg** [comment]
verified with VS 17.2.0 Preview 3.0 [32317.292.main], the Rectangle is just in Microsoft.Maui.Controls.Shapes
the following XAML code snippet  works fine but RectangleGeometry was not found in Border
```
<Border>
    <Border.StrokeShape>
        <Rectangle />
    </Border.StrokeShape>
</ Border>
```
![image](https://user-images.githubusercontent.com/20982341/158941360-f725acbc-2b12-4f0d-a136-7c018654c387.png)
demo: [MauiApp3558.zip](https://github.com/dotnet/maui/files/8301740/MauiApp3558.zip)

**Comment by whichone** [comment]
It seems that only Ellipse, Recangle, RoundRectangle can be resized properly based on the content inside. Other shapes like line, path, polyline etc, are drawn with fixed dimensions and won't be resized to hold the content as a BORDER. What's the scenario to use them then?

**Comment by Ghasan** [comment]
I don't think it needs to be a Geometry, but I understand the confusion, which needs to be fixed. Maybe into making Rectangle, RectangeShape, and RectangeGeometry?

It needs to stay as IShape so that we don't need to define size for border shape, we don't really care about the geometry of the border, but rather the outer shape, size is managed by HeightRequest and WidthRequest etc.

````xml
<Border.StrokeShape>
    <RoundRectangle CornerRadius="15,15,0,0" /> <!-- won't be possible with Geometry, as I need to define the rect dimensions then. -->
</Border.StrokeShape>
````

**Comment by jinxinjuan** [comment]
Verified with VS 17.6.0 Preview 3.0, The StrokeShape is Geometry, but property StrokeGeometry is not found in Border.
![image](https://user-images.githubusercontent.com/37974803/231967372-838f1393-b582-4212-bcd8-b799d7914f72.png)

**Comment by mattleibow** [comment]
@PureWeen  @drasticactions Related to offline discussions.

**Comment by dalux-tke** [comment]
This [issue ](https://github.com/dotnet/maui/issues/16919) was linked to here and closed. However, I am still having the issue in the closed ticket as the live preview crashes due to that error.
 
Is there any news on the state of this old issue?
